docs/features/optin.md

```diff
@@ -0,0 +1,106 @@
+
+We recommend only enabling the experimental direct controller if it is
+necessary to work around a legacy controller bug or to enable a new feature that
+is only available in the direct controller.
+
+
+## Enabling
+
+To enable the direct controller for a specific resource, update the resource and
+specify the annotation:
+
+```yaml
+metadata:
+  annotations:
+    alpha.cnrm.cloud.google.com/reconciler: "direct"
+```
+
+For example:
+
+```yaml
+apiVersion: sql.cnrm.cloud.google.com/v1beta1
+kind: SQLInstance
+metadata:
+  name: my-sqlinstance
+  annotations:
+    alpha.cnrm.cloud.google.com/reconciler: "direct"
+spec:
+  databaseVersion: POSTGRES_16
+  region: us-central1
+  settings:
+    tier: db-custom-1-3840
+```
+
+The only supported value for the annotation is "direct", to enable the
+direct controller.
+
+It's worth noting that it's (relatively) harmless to add the annotation to enable
+experimental direct controllers for resources that do not yet support an
+experimental direct controller. In this case, there is no behavior
+change. The normal (default) controller will still be used. However, as soon
+as you upgrade to a version of Config Connector where there is an
+experimental direct controller available for that resource, then the
+experimental direct controller will be enabled.
+
+To revert back to using the legacy controller, remove the annotation.
+
+
+## Verifying
+
+To verify the direct controller code is in-use for a particular resource,
+check the logs for the phrase "Running reconcile" associated with the resource.
+
+```
+$ kubectl -n cnrm-system logs pod/cnrm-controller-manager-0 | grep -e "Running reconcile" -e "my-sqlinstance"
+```
+
+You should see output similar to the following:
+
+> {"severity":"info","timestamp":"2024-11-04T21:56:28.704Z","msg":"Running reconcile","controller":"sqlinstance-controller","controllerGroup":"sql.cnrm.cloud.google.com","controllerKind":"SQLInstance","SQLInstance":{"name":"my-sqlinstance","namespace":"default"},"namespace":"default","name":"my-sqlinstance","reconcileID":"577fe58d-14eb-4e5d-9642-c9ec4b1a3137","resource":{"name":"my-sqlinstance","namespace":"default"}}
+
+If the legacy controller is enabled (instead of the new direct controller), the
+logs will show "starting reconcile" for the resource instead of "Running reconcile".
+
+> {"severity":"info","timestamp":"2024-10-30T23:47:49.726Z","logger":"sqlinstance-controller","msg":"starting reconcile","resource":{"name":"my-sqlinstance","namespace":"default"}}
+
+
+## Applicability
+
+If you're unsure if a particular resource has an experimental direct controller
+available and want to try out the new code wherever possible (or if you're just
+curious to learn more about the internals of this feature in Config Connector),
+read on.
+
+To find out if an experimental direct controller is available for a resource,
+check CRD for the resource. If the CRD specifies either of the following legacy
```

Comment:
Would be good to give instructions for checking the CRD. We can suggest either something like `kubectl get crd alloydbinstances.alloydb.cnrm.cloud.google.com -ojson | jq '.metadata.labels' ` OR inspecting https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_alloydbinstances.alloydb.cnrm.cloud.google.com.yaml.

Reply:
Alright, added some explicit instructions for checking the CRDs.